File: typescript/src/shapes/circle.ts
// function circle(radius: number): Shape {
//     return {
//         radius,
//         computeArea: function (): number {
//             return Math.PI * radius * radius
//         }
//     }
// }

// export { circle }

import { Shape } from '../shapes/shape';


class Circle implements Shape {
    private radius: number;

    constructor(radius: number) {
        this.radius = radius;
    }

    getArea(): number {
        return Math.PI * this.radius * this.radius;
    }

    draw(): void {
        console.log(`Shape printed\nIts area is ${this.getArea()}`);
    }
}

export { Circle }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Circle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Circle implements Shape {
    // public double radius;

    // public Circle(double radius) {
    //     this.radius = radius;
    // }

    // public double getArea() {
    //     return Math.PI * radius * radius;
    // }

    private double radius;

    public Circle(double radius){
        this.radius = radius;
    }

    public double getArea(){
        return Math.PI * radius * radius;
    }

    public void draw(){
        System.out.println("Shape printed\nIts area is " + getArea());
    }
}
